src/index.ts

```diff
@@ -380,6 +411,45 @@ Please refer to the Monika documentations on how to how to configure notificatio
 
     return startupMessage
   }
+
+  async checkTLSAndSaveNotifIfFail(
+    domain: string,
+    reminder: number,
+    probe: Probe,
+    notifications?: Notification[]
+  ) {
+    try {
+      await checkTLS(domain, reminder)
+    } catch (error) {
+      log.error(error.message)
+
+      if (notifications && notifications?.length > 0) {
+        await Promise.all(
```

Comment:
Why we are not saving failed notifications asynchronously?

Reply:
Thank you, nice catch. I fixed it here [
1088244](https://github.com/hyperjumptech/monika/pull/315/commits/10882445203b6203a1c603de4021f4da25e77a54).